### DISCUSSION TITLE
Does Phase4 support a reply pattern CALLBACK

### DISCUSSION THREAD

**Discussion by gnazaryan** [phax/phase4#101] [Q&A] [answered]
Hi,

I was doing a research on the phase 4 library to find out if the lib can be used in my companies product as a mean of AS4 communication.

I did try to run the phase 4 server and communicate AS4 messages with the phase4 client, but I can't seam to be able to send async AS4 callbacks.

If I set the setSendReceiptReplyPattern as EPModeSendReceiptReplyPattern.CALLBACK, this would imply the AS4 server should send the response with a separate http call to the initiating party for the non repudiation recipient, but seams the phase 4 doesn't support it.

Would you please confirm if the async callbacks with separate http callouts are supported by the phase4 and if yes some hints would be appreciate on how to use it.

thanks!

**Comment by gnazaryan** [comment, marked as answer]
@phax Would you please share some ideas, thank you!

**Reply by phax** [reply]
Hi @gnazaryan , unfortunately the CALLBACK reply pattern is currently not supported. Only synchronous responses are supported.

How would the target URL need to be determined in that case? Would that need to be hard coded in the PMode?

Anyway, I am open to PRs :)

**Reply by gnazaryan** [reply]
@phax Thanks for the reply.

Yeah, I believe the PMode would be a good spot for the target url endpoint.

**Reply by gnazaryan** [reply]
The requesting Envelop message could also contain a callback url maybe nested on the party info attributes.
